### ISSUE TITLE
fast import crashes with: "Missing < in ident string"

### ISSUE THREAD

**Issue by noschinl** [felipec/git-remote-hg#5]
Hi,

when cloning a certain mercurial repository, fast-import crashes:

```
fast-import crash report:
    fast-import process: 12990
    parent process     : 12989
    at Wed Jun 4 15:53:16 2014

fatal: Missing < in ident string: noschinl 1397474979 +0200
```

The most recent command before the crash is:

```
committer noschinl 1397474979 +0200
```

If I lookup the associated commit in Mercurial, I see this this refers to the content of the committer field in extras, which was created by an older version of git-remote-hg:

```
$ hg log -r 56572 --template "{join(extras, '\n')}\n"
branch=default
committer=noschinl 1397474979 -7200
```

So apparently, once upon a time git-remote-hg created commits that itself cannot import anymore now. This may be related to the fact that my git config contains the lines:

```
[user]
	name = noschinl
	email = unknown
```

as the project uses old unix-style author names.

I don't know whether this is the correct fix, but I made the import work for me using the following patch:

```
diff --git a/git-remote-hg b/git-remote-hg
index c10b37b..d0e94c2 100755
--- a/git-remote-hg
+++ b/git-remote-hg
@@ -498,7 +498,7 @@ def export_ref(repo, name, kind, head):
         if 'committer' in extra:
             try:
                 cuser, ctime, ctz = extra['committer'].rsplit(' ', 2)
-                committer = "%s %s %s" % (cuser, ctime, gittz(int(ctz)))
+                committer = "%s %s %s" % (fixup_user(cuser), ctime, gittz(int(ctz)))
             except ValueError:
                 cuser = extra['committer']
                 committer = "%s %d %s" % (fixup_user(cuser), time, gittz(tz))

```

The versions of relevant tools are:

$ git --version
git version 2.0.0.rc2
$ hg --version
Mercurial Distributed SCM (version 2.9.2)

git-remote-hg version is 185852eac44c25ae2e8d3b3fb6c9630e754e6363

**Comment by fingolfin** [comment]
Just for the record, I added that fix to my fork of git-remote-hg:
  https://github.com/fingolfin/git-remote-hg
Would be interesting to know if that solves the issue for you?

**Comment by noschinl** [comment]
Yes, with your master branch I can import the repository without problems.

**Comment by nomeata** [comment]
What is the status of this bug? I was just bitten by it.

**Comment by encukou** [comment]
The status is that @felipec abandoned this project but doesn't want to give ownership to anyone else.
Use @fingolfin's fork.

**Comment by nomeata** [comment]
@wRAR (← Debian package maintainer of `hg-fast-export`), have you considered switching to that branch? (Or am I confusing unrelated projects here?)

**Comment by wRAR** [comment]
@nomeata https://github.com/felipec/git/wiki/Comparison-of-git-remote-hg-alternatives (`hg-fast-export` is `fast-export` and `cosmin’s git-hg` there. `git-remote-hg` is packaged separately.

**Comment by nomeata** [comment]
Ah, thanks. I guess I made quick conclusions due to the similarity of the error messages. Sorry for the noise.

**Comment by fingolfin** [comment]
That comparison is also a bit outdated. Oh, and there is an interesting new contender: https://github.com/glandium/git-cinnabar (haven't tried it myself yet, but its author seems quite active).

**Comment by felipec** [comment]
Fixed in 517ceb9. It would have been useful if the patch was sent to the mailing list.

**Comment by felipec** [comment]
@encukou Noone abandoned anything.
